repose-aggregator/core/repose-core/src/main/java/org/openrepose/powerfilter/intrafilterLogging/RequestLog.java

```diff
@@ -38,41 +37,34 @@
     String httpMethod;
     String requestURI;
     String requestBody;
-    HashMap<String, String> headers;
-
-    public RequestLog(MutableHttpServletRequest mutableHttpServletRequest,
-                      FilterContext filterContext) throws IOException {
-
-        Filter filter = filterContext.getFilterConfig();
+    Map<String, String> headers;
 
+    public RequestLog(HttpServletRequest httpServletRequest, Filter filter) throws IOException {
         preamble = "Intrafilter Request Log";
         timestamp = new DateTime().toString();
         currentFilter = StringUtils.isEmpty(filter.getId()) ? filter.getName() : filter.getId() + "-" + filter.getName();
-        httpMethod = mutableHttpServletRequest.getMethod();
-        requestURI = mutableHttpServletRequest.getRequestURI();
-        headers = convertRequestHeadersToMap(mutableHttpServletRequest);
+        httpMethod = httpServletRequest.getMethod();
+        requestURI = httpServletRequest.getRequestURI();
+        headers = convertRequestHeadersToMap(httpServletRequest);
 
-        //Have to wrap the input stream in something that can be buffered, as well as reset.
-        ServletInputStream bin = mutableHttpServletRequest.getInputStream();
-        bin.mark(Integer.MAX_VALUE); //Something doesn't support mark reset
-        requestBody = IOUtils.toString(bin); //http://stackoverflow.com/a/309448
-        bin.reset();
+        ServletInputStream inputStream = httpServletRequest.getInputStream();
+        if (inputStream.markSupported()) {
+            inputStream.mark(Integer.MAX_VALUE);
+            requestBody = IOUtils.toString(inputStream); //http://stackoverflow.com/a/309448
+            inputStream.reset();
+        } else {
+            throw new RuntimeException("Servlet input stream does not support mark/reset: " + inputStream);
```

Comment:
I just want to make sure that we want to throw an exception here rather than, saying, logging a warning. The input stream _should_ always support mark since we pass the request in from our code. If at some point in the future we pass a request with an input stream that does not support mark, throwing an exception should make our tests bomb immediately so that we know what we did. Of course, we likely have tests verifying that the body gets logged by intra-filter logging, so that may be redundant.

Also, RuntimeException is pretty generic. Is there a better exception to use from the standard library? Maybe IllegalArgumentException? It doesn't matter too much, just a thought.

Really, I'd just like to hear your thoughts on the matter.